lm_human_preference_details/train_policy_accelerate.py

```diff
@@ -755,7 +790,8 @@ def train(args: Args):
                     )
 
         with torch.no_grad():
-            writer.add_histogram("ppo/val/ratio_hist", ratio, update)
+            if not args.deepspeed: # for some reason there is a OOM with the `writer.add_histogram`
+                writer.add_histogram("ppo/val/ratio_hist", ratio, update)
```

Comment:
FYI I was able to train 7B models in TRL with ZeRO-2 and didn't need to remove the histogram. On the other hand that was for sentiment tuning, which is less memory intensive than your application here